share/spack/csh/spack.csh

```diff
@@ -63,6 +63,47 @@ case cd:
         cd `\spack location $_sp_arg $_sp_args`
     endif
     breaksw
+case env:
+    shift _sp_args  # get rid of 'env'
+
+    set _sp_arg=""
+    [ $#_sp_args -gt 0 ] && set _sp_arg = ($_sp_args[1])
+
+    if ( "$_sp_arg" == "-h" ) then
+        \spack env -h
+    else
+        switch ($_sp_arg)
+            case activate:
+                set _sp_env_arg=""
+                [ $#_sp_args -gt 1 ] && set _sp_env_arg = ($_sp_args[2])
+
+                if ( "$_sp_env_arg" == "" || "$_sp_args" =~ "*--sh*" || "$_sp_args" =~ "*--csh*" || "$_sp_args" =~ "*-h*" ) then
+                    # no args or args contain -h/--help, --sh, or --csh: just execute
+                    \spack $_sp_flags env $_sp_args
+                else
+                    shift _sp_args  # consume 'activate' or 'deactivate'
+                    # actual call to activate: source the output
+                    eval `\spack $_sp_flags env activate --csh $_sp_args`
+                endif
+                breaksw
+            case deactivate:
+                set _sp_env_arg=""
+                [ $#_sp_args -gt 1 ] && set _sp_env_arg = ($_sp_args[2])
+
+                if ( "$_sp_env_arg" != "" ) then
+                    # with args: execute the command
+                    \spack $_sp_flags env $_sp_args
+                else
+                    # no args: source the output
+                    eval `\spack $_sp_flags env deactivate --csh`
+                endif
+                breaksw
+            default:
+                echo default
+                \spack $_sp_flags env $_sp_arg $_sp_args
```

Comment:
@tgamblin 

this should just be `\spack $_sp_flags env $_sp_args`

Otherwise it tries to invoke commands like

`spack env list list`